kolibri/plugins/user_profile/tasks.py

```diff
@@ -152,4 +156,16 @@ def mergeuser(command, **kwargs):
     job.extra_metadata["remote_user_pk"] = remote_user_pk
     job.save_meta()
     job.update_progress(1.0, 1.0)
-    local_user.delete()
+
+    # check if current user should be promoted to superuser:
+    set_as_super_user = kwargs.get("set_as_super_user")
+    if set_as_super_user:
+        # set current user as the new superuser of this device:
+        remote_user.facility.add_role(remote_user, role_kinds.ADMIN)
```

Comment:
I'd add an inner function for these two lines as they are used also if `new_superuser_id` exists. Not for the sake of DRY as they're just two lines, but to be sure mainteinance is simple in the future if changes need to be added when an user is promoted to superuser